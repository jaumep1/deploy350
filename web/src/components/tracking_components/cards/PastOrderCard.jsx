import React from 'react';
import PropTypes from 'prop-types';
import '../../../stylesheets/delivery.css';
import DeliveryCard from './DeliveryCard';

function PastOrderCard(props) {
  const {
    date, cost, shippingDetails, orderNumber, statusDescription,
    itemDetails,
    // showButton,
    count,
  } = props;
  const {
    name: itemTitle, categories: itemCategory, _id: itemLawsonNumber, image,
  } = itemDetails;

  return (
    <DeliveryCard
      date={date}
      cost={cost}
      shippingDetails={shippingDetails}
      orderNumber={orderNumber}
      width
    >
      <div className="flex flex-col h-64 p-6">
        <div className="flex flex-col items-start mb-6">
          <div className="text-xl font-semibold text-black normal-line-height">
            Delivered
            {` ${date}`}
          </div>
          <div className="text-sm text-black normal-line-height">{statusDescription}</div>
        </div>
        <div className="flex justify-start items-start gap-4">
          <img className="w-[125px] h-[125px] object-cover object-center" alt="order" src={image} />
          <div className="flex flex-col justify-start text-start">
            <div className="text-xl font-semibold text-black">
              {`${itemTitle} (x${count})`}
            </div>
            <div className="dark-grey text-xs font-semibold mt-1">{itemCategory[0]}</div>
            <div className="text-black text-xs font-semibold">
              Lawson #
              {itemLawsonNumber}
            </div>
            {/* {showButton && (
              <button type="button" className="background-red font-semibold text-white !w-28 mt-2">
                Return Order
              </button>
            )} */}
          </div>
        </div>
      </div>
    </DeliveryCard>
  );
}

PastOrderCard.propTypes = {
  count: PropTypes.number.isRequired,
  date: PropTypes.string.isRequired,
  cost: PropTypes.number.isRequired,
  shippingDetails: PropTypes.string.isRequired,
  orderNumber: PropTypes.number.isRequired,
  statusDescription: PropTypes.string.isRequired,
  itemDetails: PropTypes.shape({
    name: PropTypes.string.isRequired,
    categories: PropTypes.arrayOf(PropTypes.string).isRequired,
    _id: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
  // showButton: PropTypes.bool.isRequired,
};

export default PastOrderCard;
