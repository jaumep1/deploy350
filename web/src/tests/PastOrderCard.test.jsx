/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import {
  render, screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PastOrderCard from '../components/tracking_components/cards/PastOrderCard';

describe('PastOrderCard component', () => {
  const mockProps = {
    date: '2022-03-20',
    cost: 25.99,
    shippingDetails: '123 Main St, Anytown USA',
    orderNumber: 1234567890,
    statusDescription: 'Delivered',
    itemDetails: {
      name: 'Test Item',
      categories: ['Category 1', 'Category 2'],
      id: 987654321,
      image: 'https://www.example.com/test.jpg',
    },
    showButton: true,
  };

  test('renders PastOrderCard component with all props', () => {
    render(<PastOrderCard {...mockProps} />);
    const orderDate = screen.getByText(/Delivered 2022-03-20/i);
    const itemTitle = screen.getByText(/Test Item/i);
    const itemCategory = screen.getByText(/Category 1/i);
    const lawsonNumber = screen.getByText(/Lawson #987654321/i);
    const returnButton = screen.getByRole('button', { name: /Return Order/i });
    expect(orderDate).toBeInTheDocument();
    expect(itemTitle).toBeInTheDocument();
    expect(itemCategory).toBeInTheDocument();
    expect(lawsonNumber).toBeInTheDocument();
    expect(returnButton).toBeInTheDocument();
  });

  test('renders PastOrderCard component without return button', () => {
    const { showButton, ...rest } = mockProps;
    render(<PastOrderCard {...rest} />);
    const orderDate = screen.getByText(/Delivered 2022-03-20/i);
    const itemTitle = screen.getByText(/Test Item/i);
    const itemCategory = screen.getByText(/Category 1/i);
    const lawsonNumber = screen.getByText(/Lawson #987654321/i);
    const returnButton = screen.queryByRole('button', { name: /Return Order/i });
    expect(orderDate).toBeInTheDocument();
    expect(itemTitle).toBeInTheDocument();
    expect(itemCategory).toBeInTheDocument();
    expect(lawsonNumber).toBeInTheDocument();
    expect(returnButton).not.toBeInTheDocument();
  });

  test('clicking return button calls handler', () => {
    const mockHandler = jest.fn();
    render(<PastOrderCard {...mockProps} onReturn={mockHandler} />);
    const returnButton = screen.getByRole('button', { name: /Return Order/i });
    userEvent.click(returnButton);
    expect(mockHandler).toHaveBeenCalledTimes(0);
  });
});
