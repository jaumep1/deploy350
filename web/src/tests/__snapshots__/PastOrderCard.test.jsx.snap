// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastOrderCard renders correctly 1`] = `
<div>
  <div
    class="delivery-card w-full max-w-[700px]"
  >
    <div
      class="delivery-card-header flex justify-between h-12 px-2.5 py-1 text-xs"
    >
      <div
        class="flex items-center justify-between gap-8"
      >
        <div
          class="flex flex-col items-start gap-0.25"
        >
          <div
            class="font-semibold"
          >
            Order Placed
          </div>
          <div>
            2022-01-01
          </div>
        </div>
        <div
          class="flex flex-col items-start gap-0.25"
        >
          <div
            class="font-semibold"
          >
            Total
          </div>
          <div>
            50.00
          </div>
        </div>
        <div
          class="flex flex-col items-start gap-0.25"
        >
          <div
            class="font-semibold"
          >
            Ship To
          </div>
          <div>
            123 Main St, Anytown USA
          </div>
        </div>
      </div>
      <div
        class="flex items-center justify-between gap-2"
      >
        <div
          class="flex flex-col items-end gap-0.25 flex-end"
        >
          <div
            class="font-semibold"
          >
            Order #
            12345
          </div>
          <div>
            <u>
              View order details
            </u>
          </div>
        </div>
      </div>
    </div>
    <div
      class="min-h-[100px]"
    >
      <div
        class="flex flex-col h-64 p-6"
      >
        <div
          class="flex flex-col items-start mb-6"
        >
          <div
            class="text-xl font-semibold text-black normal-line-height"
          >
            Delivered
             2022-01-10
          </div>
          <div
            class="text-sm text-black normal-line-height"
          >
            Delivered
          </div>
        </div>
        <div
          class="flex justify-start items-start gap-4"
        >
          <img
            alt="order"
            class="w-[125px]"
            src="https://example.com/image.jpg"
          />
          <div
            class="flex flex-col justify-start text-start"
          >
            <div
              class="text-xl font-semibold text-black"
            >
              Item 1
            </div>
            <div
              class="dark-grey text-xs font-semibold mt-1"
            >
              Category A
            </div>
            <div
              class="text-black text-xs font-semibold"
            >
              Lawson #
              6789
            </div>
            <button
              class="background-red font-semibold text-white !w-28 mt-2"
              type="button"
            >
              Return Order
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
